assets/app/scripts/controllers/catalog/images.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:PodsController
+ * @description
+ * # ProjectController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('CatalogImagesController', function ($scope, DataService, $filter, LabelFilter, imageEnvFilter, $routeParams) {
+    $scope.projectImageRepos = {};
+    $scope.openshiftImageRepos = {};
+    $scope.builders = [];
+    $scope.sourceURL = $routeParams.builderfor;
+
+    var imagesForRepos = function(imageRepos, scope) {
+      // TODO this is terrible but it's the best we can do right now until we get API changes
+      angular.forEach(imageRepos, function(imageRepo) {
+        angular.forEach(imageRepo.tags, function(dockerTag, imageRepoTag) {
+          DataService.get("imageRepositoryTags", imageRepo.metadata.name + ":" + imageRepoTag, scope).then(function(image) {
+            // TODO this is how we will do this for now, long term we actually want a per tag annotation on the image repository
+            // that says it if it is a builder
+            // If the image specifies an STI_SCRIPTS_URL (so it is a builder), but doesn't have OPENSHIFT_BUILD_NAME (which gets
+            // added to the output images of our STI builds)
+            if (imageEnvFilter(image, 'STI_SCRIPTS_URL') && !imageEnvFilter(image, 'OPENSHIFT_BUILD_NAME')) {
+              var builder = {
+                imageRepo: imageRepo,
+                imageRepoTag: imageRepoTag,
+                image: image,
+                name: imageRepo.metadata.name + ":" + imageRepoTag
+              };
+              $scope.builders.push(builder);
+            }
+            console.log("builders", $scope.builders);
+          });
+        });
+      });
+    };
+
+    DataService.list("imageRepositories", $scope, function(imageRepos) {
+      $scope.projectImageRepos = imageRepos.by("metadata.name");
+      imagesForRepos($scope.projectImageRepos, $scope);
+
+      console.log("project image repos", $scope.projectImageRepos);
+    });
+
+    DataService.list("imageRepositories", {namespace: "openshift"}, function(imageRepos) {
```

Comment:
May be good to have a service that returns the default namespace ($openshiftNamespace) so it's in one place